chapters/renderpass.txt

```diff
@@ -1609,21 +1609,37 @@ attachments on multiple uses via ename:VK_ATTACHMENT_LOAD_OP_CLEAR.
   * [[VUID-VkAttachmentDescription-finalLayout-00843]]
     pname:finalLayout must: not be ename:VK_IMAGE_LAYOUT_UNDEFINED or
     ename:VK_IMAGE_LAYOUT_PREINITIALIZED
+ifndef::VK_VERSION_1_1,VK_VERSION_1_2,VK_KHR_maintenance2[]
+  * [[VUID-VkAttachmentDescription-format-03280]]
+    If pname:format is a color format, pname:initialLayout must: not be
+    ename:VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL or
+    ename:VK_IMAGE_LAYOUT_DEPTH_STENCIL_READ_ONLY_OPTIMAL
+endif::VK_VERSION_1_1,VK_VERSION_1_2,VK_KHR_maintenance2[]
+ifdef::VK_VERSION_1_1,VK_VERSION_1_2,VK_KHR_maintenance2[]
   * [[VUID-VkAttachmentDescription-format-03280]]
```

Comment:
Can we use the same VUID for the block ifndef-endif-ifdef-endif? Looks like it is working. 

I noticed your replacement, it is a little ugly because it will have some text (like `If pname:format is a color format, pname:initialLayout must: not be`) twice. And that text is a bulletin without VUID. And I also noticed a following PR from someone set a VUID-TODO for it. 

If this solution is OK. It resolved issues in those two PRs. 

BTW, I also added VK_VERSION_1_2 in ifs, otherwise, it is incorrect for VK 1.2 specification. But this turned out that if we have VK 1.3 or VK 2.0, all these following versions need to be added into the list. Is there a solution like if VK_VERSION_1_1+, which means VK 1.1 and all versions above?

Reply:
What you originally did with this PR was to recognize that a couple of the enums in the VUs were not valid without the extension and/or version that introduced those enums defined, and that was a good observation. I re-did the PR in #1708 to account for the limitations of how the validusage statement extractor actually works. The changes you've added to this PR since then are not consistent with how our process works, and we can't accept those. They don't prevent generating a spec but they will cause problems with actually using the VUs in the validation layers process downstream.

 I believe #1708 captures the intent of this PR and would like to go with that one, unless you see something actually wrong (in contrast to appearing inelegant or wordy) with it.

> Can we use the same VUID for the block ifndef-endif-ifdef-endif? Looks like it is working.

No, that's not OK because they are two different valid usage conditions being tested. This doesn't affect generating the spec outputs but it will be a problem for validation layer implementation.

> I noticed your replacement, it is a little ugly because it will have some text (like If pname:format is a color format, pname:initialLayout must: not be) twice.

Yes, it's unfortunate that there is some repetition. But as I commented previously, the valid usage extractor has restrictions that force this mild ugliness on us.

>  And that text is a bulletin without VUID.

In the PR, it does not have a VUID yet. VUIDs are assigned to new VU statements as part of the public spec update process, which is the same point at which the valid usage database validusage.json is generated for consumption by the validation layers. This is a documented part of the process of managing VU statements.

> BTW, I also added VK_VERSION_1_2 in ifs, otherwise, it is incorrect for VK 1.2 specification.

What the original conditional was saying was that if you build a spec with VK_VERSION_1_1 defined, or without VK_VERSION_1_1 but with VK_KHR_maintenance2 (e.g Vulkan 1.0 + maintenance2 extension), then the ifdefed content will be included. VK_VERSION_1_2 by definition requires VK_VERSION_1_1. We will never generate a spec where this is not true. The ifdef needs to contain only the extensions and/or versions where the content is actually first defined.

Reply:
@oddhack , I think as a senior folk and a reviewer as you in Vulkan community, it might make more sense to give insights and suggestions on the other two issues at #1700 and #1699, and I really appreciate it if you can do that. They are much more difficult and need real discussions. While polishing this PR is really trivial. And it's my duty.  

For the trivial stuff like fulfilling the requirement of the autogen/infrastructure for Vk spec, I think you can guide me to do that. And that is what a reviewer should do. According to my experience, the reviewer should not submit a new PR/change to compete with the original author unless the original author meet significant hard-to-resolve problem by himself/herself or the original author disappears and doesn't response to the thread. Otherwise, if the reviewer see some problem, he just submits a replacement and ask the original author to close his original PR, I feel like it is not a friendly way to promote this project. 

Regarding the VUID stuff, thanks for your explanation! I revised this PR accordingly. It can handle the ugly repetition problem. And we can assign new VUID afterwards. 

Regarding the VK_VERSION_1_2 problem, it turned out that I used `incorrect` way to generate the html. I tried `./makeSpec -spec core html VERSIONS="VK_VERSION_1_2" `. It should be
`./makeSpec -spec core html VERSIONS="VK_VERSION_1_0 VK_VERSION_1_1 VK_VERSION_1_2" ` or just `./makeSpec -spec core html` for now. Likewise, if we want to generate VK 1.1, we need to add VK 1.0 on the list. You see, if Vulkan gets more and more versions, we need to add all previous versions to generate a particular version spec. That's a little silly. I think the autogen tool can expand versions for a particular version via script. However, I understand that the infrastructure to autogen of Vk spec is only used by a very small amount of people, so it is up to you guys whether to do that or not. 

Once again, I really appreciate if you can comment on issue 1700 and 1699, Jon. I really want to hear the thoughts from senior folks like you in Vulkan community.

Reply:
I am senior in terms of length of time involved, but that is far from making me an expert on everything in the specification. #1699 has several responses from WG members who are more familiar with it already, and #1700 is being discussed internally by the Working Group and will have a reply posted when there is more information from IHVs available.

There's no problem accepting this PR once it gets to the right place in terms of the markup, but it is not quite there yet. I had not heard back for nearly a week prior to posting #1708 so made that to move things along, but I'm glad you've been able to engage again here.

Reply:
Understood. Vulkan is so complicated. It is virtually impossible for a single person to know every details even as senior as you are in the community. And I really thank you for your feedback about 1700 and 1699. Regarding 1699, I will do more investigation about the implementation in the engine I am working on. I am not convinced. And I will reply on that issue. But I am glad to know that you folks have active follow-ups on these issues, especially for issue 1700. 

Regarding the delay, I am sorry for that. It was Thanksgiving holiday, and I had a couple more day off last week. But I will definitely work on this PR.